WordPressKit/AccountSettingsRemote.swift

```diff
@@ -172,10 +171,12 @@ public class AccountSettingsRemote: ServiceRemoteWordPressComREST {
             let webAddress = response["user_URL"] as? String,
             let language = response["language"] as? String,
             let tracksOptOut = response["tracks_opt_out"] as? Bool,
-            let blockEmailNotifications = response["subscription_delivery_email_blocked"] as? Bool else {
-                WPKitLogError("Error decoding me/settings response: \(responseObject)")
-                throw ResponseError.decodingFailure
-            }
+            let blockEmailNotifications = response["subscription_delivery_email_blocked"] as? Bool
+        else {
+            WPKitLogError("Error decoding me/settings response: \(responseObject)")
+            throw ResponseError.decodingFailure
+        }
+        let twoStepEnabled = response["two_step_enabled"] as? Bool ?? false
```

Comment:
👋 Hey @salimbraksa - I was curious why this property is outside of the `guard` - are there instances where `two_step_enabled` isn't sent?

Reply:
@twstokes Honestly I don't know if `two_step_enabled` is optional ( on the Backend side ). If the property is inside the `guard-let` and the property turns out to be `nil`, the whole request will fail and this change will cause a regression bug.

So to be safe, I'm handling that mapping outside the `guard` and I believe this won't cause any regression issues whether `two_step_enabled` is optional or not.

But out of curiosity, how can we know whether `two_step_enabled` is optional or not on the Backend side?

Reply:
👋 That makes sense. From what I can tell, it won't be optional:

- We specify version 1.1 of the public API [here](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/474ea9cda1ec1b35e28f0bd967c46ec145405548/WordPressKit/AccountSettingsRemote.swift#L33).
- The [endpoint documentation](https://developer.wordpress.com/docs/api/1.1/get/me/settings/) doesn't mention `two_step_enabled` being optional or any different than the other parameters.
- You can also check out the internal PHP class for the endpoint (I'll provide the link).

Perhaps we should make it uniform with the other parameters unless we're sure it may not always be sent for v1.1 on .com. What do you think about that?

Reply:
Agreed. I'll make the changes. Thanks for helping with this. 🙏

Reply:
Resolved in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/567/commits/bbaf2dca6825c47c6a512f5cb43640604f80d224